rewrite-maven/src/main/java/org/openrewrite/maven/tree/ResolvedPom.java

```diff
@@ -126,6 +131,12 @@ public ResolvedPom(Pom requested, Iterable<String> activeProfiles) {
     @Nullable // on older LSTs, this field is not yet present
     List<String> subprojects = emptyList();
 
+    public Map<String, String> getProperties() {
```

Comment:
How often is this called? can we cache the result to prevent creating a new HashMap everytime?

Reply:
Yeah, this is the one part I am not happy with 😞. Maybe we can cache it, but I fear that would lead to another can of worms when we forget to update the cache. So that's why I just left it here like this.